--- a/com/mojang/blaze3d/platform/MacosUtil.java
+++ b/com/mojang/blaze3d/platform/MacosUtil.java
@@ -19,8 +_,16 @@
     private static final int NS_RESIZABLE_WINDOW_MASK = 8;
     private static final int NS_FULL_SCREEN_WINDOW_MASK = 16384;
 
+    public static boolean isInNativeFullscreen(long windowHandle) {
+        return getNsWindow(windowHandle).filter(MacosUtil::isInNativeFullscreen).isPresent();
+    }
+
     public static void exitNativeFullscreen(Window window) {
         getNsWindow(window).filter(MacosUtil::isInNativeFullscreen).ifPresent(MacosUtil::toggleNativeFullscreen);
+    }
+
+    public static void enterNativeFullscreen(Window window) {
+        getNsWindow(window).filter(nsWindow -> !isInNativeFullscreen(nsWindow)).ifPresent(MacosUtil::toggleNativeFullscreen);
     }
 
     public static void clearResizableBit(Window window) {
